app/models/mark.rb

```diff
@@ -0,0 +1,102 @@
+
+class Mark < ActiveRecord::Base
+  has_and_belongs_to_many :spies
+  has_many :media
+
+  validates :username, :provider, :uid, presence: true
+  validates :uid, uniqueness: true
+
+  def self.vimeo_lookup(search_term)
+    auth = "Bearer #{ENV['VIMEO_ACCESS_TOKEN']}"
```

Comment:
It's cleaner to separate the communication with the API from the Model itself. The Model's role is to manipulate and store the objects in your DB, and it would be less entangled for it to call an API-interaction class rather than incorporating the API calls directly into the model.